src/utils/convert.hpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CONVERT_HPP_
+#define CONVERT_HPP_
+
+#include <aos/common/cloudprotocol/alerts.hpp>
+#include <aos/common/cloudprotocol/envvars.hpp>
+#include <aos/common/cloudprotocol/log.hpp>
+#include <aos/common/monitoring/monitoring.hpp>
+#include <aos/common/types.hpp>
+#include <aos/sm/networkmanager.hpp>
+
+#include <servicemanager/v4/servicemanager.grpc.pb.h>
+
+namespace aos::sm::utils {
+
+/**
+ * Converts byte array to aos.
+ *
+ * @param arr array to convert.
+ * @return const Array<uint8_t>.
+ */
+const Array<uint8_t> ConvertByteArrayToAos(const std::string& arr);
+
+/**
+ * Converts aos error to protobuf error.
+ *
+ * @param error aos error.
+ * @return iamanager::v5::ErrorInfo.
+ */
+::common::v1::ErrorInfo ConvertAosErrorToProto(const Error& error);
+
+/**
+ * Converts aos instance ident to protobuf.
+ *
+ * @param src instance ident to convert.
+ * @param[out] dst protobuf instance ident.
+ * @return void.
+ */
+void ConvertToProto(const InstanceIdent& src, ::common::v1::InstanceIdent& dst);
+
+/**
+ * Converts aos push log to protobuf.
+ *
+ * @param src push log to convert.
+ * @param[out] dst protobuf log data.
+ * @return void.
+ */
+void ConvertToProto(const cloudprotocol::PushLog& src, ::servicemanager::v4::LogData& dst);
+
+/**
+ * Converts aos monitoring data to protobuf.
+ *
+ * @param src monitoring data to convert.
+ * @param timestamp monitoring data timestamp.
+ * @param[out] dst protobuf monitoring data.
+ * @return void.
+ */
+void ConvertToProto(
+    const monitoring::MonitoringData& src, const Time& timestamp, ::servicemanager::v4::MonitoringData& dst);
+
+/**
+ * Converts aos node monitoring data to protobuf.
+ *
+ * @param src aos node monitoring.
+ * @param[out] dst protobuf message.
+ * @return void.
+ */
+template <typename T>
+void ConvertToProto(const monitoring::NodeMonitoringData& src, T& dst)
```

Comment:
Why template is required?

Reply:
Protobuf InstantMonitoring and AverageMonitoring  messages has the same payloade.
Refactored helper to accept payload (2 out args), so template is not needed